src/Playwright/Transport/WebSocketTransport.cs

```diff
@@ -0,0 +1,194 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Playwright.Transport
+{
+    internal class WebSocketTransport : IConnectionTransport, IDisposable
+    {
+        private const int DefaultBufferSize = 1024;  // Byte buffer size
+        private readonly ClientWebSocket _webSocket;
+        private readonly string _wsEndpoint;
+        private readonly CancellationTokenSource _readerCancellationSource = new();
+        private readonly BrowserTypeConnectOptions _options;
+
+        internal WebSocketTransport(
+            string wsEndpoint = default,
+            BrowserTypeConnectOptions options = default)
+        {
+            _webSocket = new ClientWebSocket();
+            _wsEndpoint = wsEndpoint;
+            _options = options;
+            SetHeaders();
+            _ = ConnectAsync();
+        }
+
+        /// <inheritdoc cref="IDisposable.Dispose"/>
+        ~WebSocketTransport() => Dispose(false);
+
+        public event EventHandler<MessageReceivedEventArgs> MessageReceived;
+
+        public event EventHandler<LogReceivedEventArgs> LogReceived;
+
+        public event EventHandler<TransportClosedEventArgs> TransportClosed;
+
+        public bool IsClosed { get; private set; }
+
+        public async Task SendAsync(string message)
+        {
+            var messageBuffer = Encoding.UTF8.GetBytes(message);
+            var messagesCount = (int)Math.Ceiling((double)messageBuffer.Length / DefaultBufferSize);
```

Comment:
Style: You would typically implement this as following:
```cs
for (var offset = 0; offset < messageBuffer.length; offset += DefaultBufferSize) {
    var count = Math.Min(messageBuffer.length - offset, messageBuffer.length);
    new ArraySegment<byte>(messageBuffer, offset, count);
}
```

Reply:
How would I identify `endOfMessage` here? 
`Count <= 0` ?

Reply:
There was no endOfMessage in your snippet, I'm just proposing a shorter and a more standard alternative. For loop will end when the message is sent.

Reply:
I tried breaking few cases. I will try to refactor it later.

Reply:
have fixed this